### ISSUE TITLE
Unified Framework for Registration and Segmentation of the Spinal Cord and Lesions

### ISSUE THREAD

**Issue by naga-karthik** [neuropoly/idea-projects#21]
**Motivation:** We usually have datasets with several contrasts but typically only the ground-truth for a single contrast. To get around this, we usually co-register the contrasts, copy the same label for each contrast and then train the model. This co-registration step is usually done independently requires some expertise in image registration (which cannot always be assumed from a user perspective)

**Idea:** Develop a unified approach that performs the co-registration between contrasts and trains a segmentation model (by concatenating the co-registered contrasts as input). 

**Approach:** 
* **Step 1.** Use SynthMorph approach that already works reasonably for co-registration and it is already implemented in SCT. 
* **Step 2.** Use the synthmorph output as the inputs to the segmentation netowork. There will be two loss functions: (1) for learning the registration properly, and (2) for learning the segmentation based off of the co-registered inputs. 

As this is a common problem in medical imaging, there are definitely some works that have done this. Will update the issue once when I find them. 

Note: This, for now, is the high-level idea for the project. The approach will surely change in the future!

**Comment by simonqueric** [comment]
related to https://github.com/ivadomed/ivadomed/issues/659

**Comment by jcohenadad** [comment]
Duplicate of https://github.com/ivadomed/ivadomed/issues/659
